src/App.tsx

```diff
@@ -1,26 +1,506 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  deleteTodo,
+  getTodos,
+  addTodo,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Error } from './components/ErrorMessage/ErrorMessage';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [query, setQuery] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [itemsLeft, setItemsLeft] = useState<number>(0);
+  const [sortBy, setSortBy] = useState('all');
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [deleteTodoId, setDeleteTodoId] = useState<number | null>(null);
+  const [isLoadingChange, setIsLoadingChange] = useState(false);
+  const [cleanCompleted, setCleanCompleted] = useState<boolean>(false);
+  const [newTask, setNewTask] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isSubmiting, setIsSubmiting] = useState<boolean>(false);
+  const [isUpdating, setIsUpdating] = useState<number | null>(null);
+  const [itemEditingId, setItemEditingId] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState<string>('');
+  const [todosLength, setTodosLength] = useState<number>(0);
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
```

Comment:
You don't need todos and allTodos states. Keep only one (name `todos`, but store all todos there) state and compute filtered todos based on other states and save filtered array into a variable

Also do filtering inside useMemo callback